### PR TITLE
add a complex type de-serialization test

### PR DESCRIPTION
**Pull request by yannicklamprecht**

### CODE REVIEW

File: src/test/java/de/eldoria/jacksonbukkit/deserializer/ComplexPojoDeserializerTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+public class ComplexPojoDeserializerTest implements SerializationTest {
+    static UUID worldUID = new UUID(0, 0);
+
+    @BeforeAll
+    static void setup() {
+        var serverMock = spy(MockBukkit.mock());
```
Comment (yannicklamprecht) [inline src/test/java/de/eldoria/jacksonbukkit/deserializer/ComplexPojoDeserializerTest.java:28]: spy can likely be removed here.

File: src/test/java/de/eldoria/jacksonbukkit/serializer/ComplexPojoSerializerTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+public class ComplexPojoSerializerTest implements SerializationTest {
+
+    static UUID worldUID = new UUID(0, 0);
+
+    @BeforeAll
+    static void setup() {
+        var serverMock = spy(MockBukkit.mock());
```
Comment (yannicklamprecht) [inline src/test/java/de/eldoria/jacksonbukkit/serializer/ComplexPojoSerializerTest.java:29]: py can likely be removed here.

File: src/test/java/de/eldoria/jacksonbukkit/templates/ComplexPojoTemplate.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      ColorTemplate.SINGLE,
+            FireworkEffectTemplate.SINGLE,
+            LocationTemplate.SINGLE,
+            NamespacedKeyTemplate.SINGLE,
+            PatternTemplate.SINGLE,
+            PotionEffectTemplate.SINGLE,
+            VectorTemplate.SINGLE
+    );
+
+
```
Comment (yannicklamprecht) [inline src/test/java/de/eldoria/jacksonbukkit/templates/ComplexPojoTemplate.java:34]: reduce empty lines.

File: src/test/java/de/eldoria/jacksonbukkit/templates/ComplexPojoTemplate.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import org.bukkit.util.Vector;
+
+public class ComplexPojoTemplate {
+
+    public static final ComplexPojo SINGLE = new ComplexPojo(
+      AttributeModifierTemplate.SINGLE,
+      BlockVectorTemplate.SINGLE,
+      BoundingBoxTemplate.SINGLE,
+      ColorTemplate.SINGLE,
+            FireworkEffectTemplate.SINGLE,
```
Comment (rainbowdashlabs) [inline src/test/java/de/eldoria/jacksonbukkit/templates/ComplexPojoTemplate.java:26]: Reformat?

**Review by rainbowdashlabs — CHANGES_REQUESTED** [review]
Deletion of the empty files would be nice too uwu

**Review by rainbowdashlabs — APPROVED** [review]

**Pull request merged**